docs/interactivity.md

```diff
@@ -0,0 +1,457 @@
+                        "description": "subscribes a channel to greet new users",
+                        "call": {
+                            "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/subscribe",
+                            "values": {
+                                "mode": "on"
+                            }
+                        }
+                    },
+                    {
+                        "label": "unsubscribe",
+                        "hint": "[--channel]",
+                        "description": "unsubscribes a channel from greeting new users",
+                        "call": {
+                            "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/subscribe",
+                            "values": {
+                                "mode": "off"
+                            }
+                        }
+                    }
+                ]
+            }
+        ]
+    }
+]
+```
+</details>
+
+## Clicked Post Dropdown
+
+<details><summary>Request from Browser</summary>
+
+`POST` http://localhost:8065/plugins/com.mattermost.apps/api/v1/call
+
+```json
+{
+    "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/send",
+    "type": "form",
+    "expand": {
+        "app": "",
+        "acting_user": "",
+        "post": "All"
+    },
+    "context": {
+        "app_id": "hello",
+        "location_id": "send",
+        "team_id": "gzt3wbyu4tnqfmiqchyatzkx1r",
+        "channel_id": "b77x9mu8xindzxkgem8guadsra",
+        "post_id": "q389yjyqfpygtedmknkannck8c",
+        "root_id": ""
+    }
+}
+```
+
+</details>
+
+<details><summary>Response to Browser</summary>
+
+```json
+{
+    "type": "form",
+    "form": {
+        "title": "Send a survey to user",
+        "header": "Message modal form header",
+        "footer": "Message modal form footer",
+        "fields": [
+            {
+                "name": "userID",
+                "type": "user",
+                "description": "User to send the survey to",
+                "label": "User",
+                "hint": "enter user ID or @user",
+                "position": 1,
+                "modal_label": "User"
+            },
+            {
+                "name": "message",
+                "type": "text",
+                "is_required": true,
+                "value": "Here is the value of your post",
+                "label": "message",
+                "hint": "Anything you want to say",
+                "modal_label": "Text",
+                "subtype": "textarea",
+                "min_length": 2,
+                "max_length": 1024
+            }
+        ]
+    }
+}
+```
+
+</details>
+
+<details><summary>Request to App server</summary>
+
+`POST` http://localhost:8065/plugins/com.mattermost.apps/hello/send
+
+```json
+{
+    "url": "http://localhost:8065/plugins/com.mattermost.apps/hello/send",
+    "type": "form", // Notice type == "form". This means it is a request to fetch a form's definition.
```

Comment:
@larkox made a point [here](https://community-daily.mattermost.com/core/pl/9om7ewtszjrhijaj5itkrnjmfa) that this is forcing a convention onto the App developer. We should instead let the developer decide how they should structure their app, and not force them to use the same endpoint to fetch and submit a form.